toolkits/springernature/packages/springer-nature-modal/js/modal.js

```diff
@@ -0,0 +1,113 @@
+class Modal {
+	constructor(modalElement) {
+		if (!modalElement || !modalElement.hasAttribute('id')) {
+			console.error('Modal must be a valid DOM node with an ID.');
+			return;
+		}
+		this.modal = modalElement;
+		this.modalId = this.modal.getAttribute('id');
+		this.trigger = document.querySelector(`[data-modal-for='${this.modalId}']`);
```

Comment:
Just a thought - might be quite useful to allow for multiple triggers? With a querySelectorAll..
had this come up the other day when creating multiple invite buttons for a single modal

Reply:
Good idea :+1: